packages/typegpu-react/src/use-uniform-value.ts

```diff
@@ -1,13 +1,55 @@
 import type * as d from 'typegpu/data';
+import { useRoot } from './root-context.tsx';
+import { useEffect, useMemo, useState } from 'react';
+import type { ValidateUniformSchema } from 'typegpu';
 
 interface UniformValue<TSchema, TValue extends d.Infer<TSchema>> {
   schema: TSchema;
-  value: TValue;
+  value: TValue | undefined;
+  readonly $: d.InferGPU<TSchema>;
 }
 
-export function useUniformValue<TSchema, TValue extends d.Infer<TSchema>>(
-  schema: d.AnyWgslData,
+export function useUniformValue<
+  TSchema extends d.AnyWgslData,
+  TValue extends d.Infer<TSchema>,
+>(
+  schema: ValidateUniformSchema<TSchema>,
   initialValue?: TValue | undefined,
 ): UniformValue<TSchema, TValue> {
-  // TODO: Implement
+  const root = useRoot();
+
+  const [uniformBuffer] = useState(() => {
+    return root.createUniform(
+      schema,
+      initialValue,
+    );
+  });
+
+  useEffect(() => {
+    return () => {
+      uniformBuffer.buffer.destroy();
+    };
+  }, [uniformBuffer]);
```

Comment:
I don't think this will work in strict mode.

```suggestion
  const ref = useRef<number>();
  useEffect(() => {
    if (ref.current) {
      // Cancel deallocation
      clearTimeout(ref.current);
    }
    
    return () => {
      // Schedule deallocation
      ref.current = setTimeout(() => {
        uniformBuffer.buffer.destroy();
      }, 200);
    };
  }, [uniformBuffer]);
```

Reply:
You're right, I missed it.